tools/SourceKit/lib/SwiftLang/SwiftASTManager.cpp

```diff
@@ -1223,26 +1223,45 @@ ASTBuildOperationRef ASTProducer::getBuildOperationForConsumer(
     SwiftASTConsumerRef Consumer,
     IntrusiveRefCntPtr<llvm::vfs::FileSystem> FileSystem,
     SwiftASTManagerRef Mgr) {
+  ASTBuildOperationRef LatestMatch;
   for (auto &BuildOp : llvm::reverse(BuildOperations)) {
     if (BuildOp->isCancelled()) {
       continue;
     }
-    std::vector<ImmutableTextSnapshotRef> Snapshots;
-    Snapshots.reserve(BuildOp->getFileContents().size());
-    for (auto &FileContent : BuildOp->getFileContents()) {
-      if (FileContent.Snapshot) {
-        Snapshots.push_back(FileContent.Snapshot);
-      }
-    }
+
+    // No point checking for a match, we already have one - we're just looking
+    // for a finished operation that matches at this point (which we will
```

Comment:
Technically we aren’t looking for a finished operation that matches (which would imply `matchesSourceState` to me but one that can be used with snapshots (implying `canUseASTWithSnapshots`).

```suggestion
    // for a finished operation that can be used with the file contents of 'BuildOp' at this point (which we will
```

Reply:
Matches implies either to me, but I can update the comment 👍

Reply:
I just wanted to clarify this because matches is so similar to `matchesSourceState` that you can easily think that it doesn’t include `canUseASTWithSnapshots`.